From ed554b07e1849522cf468e8799bdb2aa5b6834eb Mon Sep 17 00:00:00 2001
From: Sam Thursfield <sam.thursfield@codethink.co.uk>
Date: Fri, 5 Jun 2026 14:37:28 +0100
Subject: [PATCH 17/20] FDSDK 26.08: gnomeos-deps/libtpms.bst: Patch to fix
 compile error

This was taken from the upstream 'master' branch. It's not yet in any
released version.
---
 elements/gnomeos-deps/libtpms.bst             |  2 ++
 ...ilation-error-in-TPMLIB_GetPlainText.patch | 31 +++++++++++++++++++
 2 files changed, 33 insertions(+)
 create mode 100644 patches/libtpms/Fix-a-compilation-error-in-TPMLIB_GetPlainText.patch

diff --git a/elements/gnomeos-deps/libtpms.bst b/elements/gnomeos-deps/libtpms.bst
index 0435d160c..89c494eaa 100644
--- a/elements/gnomeos-deps/libtpms.bst
+++ b/elements/gnomeos-deps/libtpms.bst
@@ -5,6 +5,8 @@ sources:
   url: github:stefanberger/libtpms.git
   track: 'v*'
   ref: v0.10.2-0-g03ff2481e133540be3b3ffe3daa1483d2a73d967
+- kind: patch
+  path: patches/libtpms/Fix-a-compilation-error-in-TPMLIB_GetPlainText.patch
 
 build-depends:
 - freedesktop-sdk.bst:public-stacks/buildsystem-autotools.bst
diff --git a/patches/libtpms/Fix-a-compilation-error-in-TPMLIB_GetPlainText.patch b/patches/libtpms/Fix-a-compilation-error-in-TPMLIB_GetPlainText.patch
new file mode 100644
index 000000000..48cbeb5d7
--- /dev/null
+++ b/patches/libtpms/Fix-a-compilation-error-in-TPMLIB_GetPlainText.patch
@@ -0,0 +1,31 @@
+commit fc8820cfaa8b5e17328f731df93911f6ab92443b
+Author: Stefan Berger <stefanb@linux.ibm.com>
+Date:   Fri Jan 2 11:37:31 2026 -0500
+
+    Fix a compilation error in TPMLIB_GetPlaintext
+    
+    Fix a compilation error that newer gcc versions may complain about:
+    
+    tpm_library.c: In function 'TPMLIB_GetPlaintext':
+    tpm_library.c:441:11: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
+      441 |     start = strstr(stream, starttag);
+          |           ^
+    At top level:
+    cc1: note: unrecognized command-line option '-Wno-self-assign' may have been intended to silence earlier diagnostics
+    cc1: all warnings being treated as errors
+    
+    Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>
+
+diff --git a/src/tpm_library.c b/src/tpm_library.c
+index f48f4fd3..7b2ea687 100644
+--- a/src/tpm_library.c
++++ b/src/tpm_library.c
+@@ -435,7 +435,7 @@ static unsigned char *TPMLIB_GetPlaintext(const char *stream,
+                                           const char *endtag,
+                                           size_t *length)
+ {
+-    char *start, *end;
++    const char *start, *end;
+     unsigned char *plaintext = NULL;
+ 
+     start = strstr(stream, starttag);
-- 
2.54.0

